src/app/dspace/dspace.directory.ts

```diff
@@ -126,30 +95,70 @@ export class DSpaceDirectory {
     }
 
     /**
-     * Method to load context details.
-     * Calls prepare with the context received.
+     * Method to load context details. Calls prepare with the context received.
      *
      * @param type
      *      string: community, collection, or item
      * @param id
-     *      current context id which needing to load context details.
+     *      current context id which needing to load context details
+     * @param page
+     *      current context page
+     * @param limit
+     *      current context limit
      */
-    loadObj(type, id, page) {
+    loadObj(type, id, page?, limit?) {
         // needed to be used within scope of promise
         let directory = this;
         return new Promise(function (resolve, reject) {
-            let context;
-            if ((context = directory.dspaceStore.get(directory.dspaceConstants[type].PLURAL, id))) {
-                directory.page(context, page);
-                directory.prepare(null, context);
-                resolve(context);
+            let parent;
+            let useCachedContext = false;
+            let directoryContext = directory.find(type, id);
+            if(directoryContext) {
+                useCachedContext = true;
+                if(type == 'item' && !directoryContext.fullItem) {
+                    useCachedContext = false;
+                }
+            }
+            if (useCachedContext) {
+                parent = directoryContext.type == 'item' ? directoryContext.parentCollection : directoryContext.parentCommunity;
+                if(limit) directoryContext.limit = limit;
```

Comment:
I'd say it's maybe a better idea to keep using brackets, even around one-liners. It appeared somewhere else in the code as well. Just because of readability, and because in the future we might want to extend the 'if' anyway.
It's a subjective matter though I know, but it might be good to consider a style where we always use the brackets to avoid possible confusion?